src/internal/m365/collection/drive/collections.go

```diff
@@ -72,6 +72,70 @@ func NewCollections(
 	}
 }
 
+func deserializeAndValidateMetadata(
+	ctx context.Context,
+	cols []data.RestoreCollection,
+) (map[string]string, map[string]map[string]string, bool, error) {
+	deltas, prevs, canUse, err := DeserializeMetadata(ctx, cols)
+	if err != nil || !canUse {
+		return deltas, prevs, canUse, clues.Stack(err).OrNil()
```

Comment:
should we always return false for `canUse`? Or are we also guaranteeing that the caller will always exit if err != nil?

Reply:
fair, might as well